base/rational.jl

```diff
@@ -140,10 +140,23 @@ Bool(x::Rational) = x==0 ? false : x==1 ? true :
 (::Type{T})(x::Rational) where {T<:Integer} = (isinteger(x) ? convert(T, x.num)::T :
     throw(InexactError(nameof(T), T, x)))
 
-AbstractFloat(x::Rational) = (float(x.num)/float(x.den))::AbstractFloat
-function (::Type{T})(x::Rational{S}) where T<:AbstractFloat where S
-    P = promote_type(T,S)
-    convert(T, convert(P,x.num)/convert(P,x.den))::T
+function numerator_denominator_promoted(x)
+    y = unsafe_rational(numerator(x), denominator(x))
+    (numerator(y), denominator(y))
+end
```

Comment:
The approach is meant to account for uncommon, but supported, types like `Rational{Union{Bool,Int}}`. I'll look into this again, though, perhaps it could be improved.

Reply:
I see, I didn't think about that case!

Reply:
Actually, this turns out to be unnecessary because the numerator and denominator get promoted before the `Rational` is constructed.